### PR TITLE
Test gas management simplification

### PR DESCRIPTION
**Pull request by alyn509**
adjusted a couple of tests on gas metering

### CODE REVIEW

**Comment by matei-p** [comment]
Can't we use constants and / or constant operations instead of numeric values in that test? e.g. instead of 1000, used gasProvied - gasUsed /* 3000 - 2000 */

**Review by matei-p — COMMENTED** [review]
see my global comment

File: arwen/contexts/metering_test.go [hunk 1]
```diff
@@ -279,80 +282,212 @@ func TestMeteringContext_GasUsed_NoStacking(t *testing.T) {
 	require.Equal(t, uint64(1401), gasUsedByContract)
 }
 
-func TestMeteringContext_GasUsed_StackOneLevel(t *testing.T) {
-	t.Skip()
+func setUpStackOneLevel(t *testing.T) (*meteringContext, *outputContext, *contextmock.RuntimeContextMock, *vmcommon.ContractCallInput, uint64, *vmcommon.ContractCallInput) {
```
Comment (camilbancioiu) [inline arwen/contexts/metering_test.go:285]: Suggest splitting this setup function into smaller ones. It returns far too many things.

For example, returning the mocked `host` already provides access to its inner contexts, so you won't have to return them individually.

File: arwen/contexts/metering_test.go [hunk 2]
```diff
@@ -279,80 +282,212 @@ func TestMeteringContext_GasUsed_NoStacking(t *testing.T) {
...
+
 	contractSize := uint64(1000)
 	contract := make([]byte, contractSize)
+
 	mockRuntime.SCCodeSize = contractSize
 	mockRuntime.SCAddress = []byte("parent")
 
 	mockRuntime.SetPointsUsed(0)
-	parentInput := &vmcommon.ContractCallInput{VMInput: vmcommon.VMInput{}}
+	parentInput := &vmcommon.ContractCallInput{
```
Comment (camilbancioiu) [inline arwen/contexts/metering_test.go:304]: Consider receiving `parentInput` and `childInput` as parameters in this function, provided by whatever test calls it.

File: arwen/contexts/metering_test.go [hunk 3]
```diff
@@ -279,80 +282,212 @@ func TestMeteringContext_GasUsed_NoStacking(t *testing.T) {
...
 	metering.PushState()
 	metering.InitStateFromContractCallInput(&childInput.VMInput)
-	require.Equal(t, uint64(500), metering.initialGasProvided)
+	require.Equal(t, 500, int(metering.initialGasProvided))
 
+	return metering, output, mockRuntime, parentInput, parentPointsBeforeStacking, childInput
+}
+
+func TestMeteringContext_GasUsed_StackOneLevel(t *testing.T) {
+	metering, output, mockRuntime, parentInput, parentPointsBeforeStacking, _ := setUpStackOneLevel(t)
```
Comment (camilbancioiu) [inline arwen/contexts/metering_test.go:347]: Consider instantiating `parentInput` here, then pass it to `setUpStackOneLevel()`.

File: arwen/contexts/metering_test.go [hunk 4]
```diff
@@ -279,80 +282,212 @@ func TestMeteringContext_GasUsed_NoStacking(t *testing.T) {
...
 	metering.UseGas(50)
-	require.Equal(t, uint64(2398), metering.GasLeft())
+	require.Equal(t, 2400, int(metering.GasLeft()))
+
+	gasSpentByContract = metering.GasSpentByContract()
+	require.Equal(t, 1600, int(gasSpentByContract))
+
+	metering.UpdateGasStateOnSuccess(output.outputState)
+
+	gasUsed := output.outputState.OutputAccounts["parent"].GasUsed
```
Comment (camilbancioiu) [inline arwen/contexts/metering_test.go:383]: Consider using the `VMOutputVerifier` here:

```
verify := NewVMOutputVerifier(t, output.outputState, nil)
verify.GasUsed("parent", 1450)
verify.GasUsed("child", 150)
verify.GasRemaining(gasRemaining)
```

File: arwen/contexts/metering_test.go [hunk 5]
```diff
@@ -279,80 +282,212 @@ func TestMeteringContext_GasUsed_NoStacking(t *testing.T) {
...
+	require.Equal(t, 1450, int(gasUsed))
+
+	gasUsed = output.outputState.OutputAccounts["child"].GasUsed
+	require.Equal(t, 150, int(gasUsed))
+
+	gasRemaining = math.SubUint64(parentInput.GasProvided, gasSpentByContract)
+	// calculate gas remaining
+
+	require.Equal(t, int(gasRemaining), int(output.GetVMOutput().GasRemaining))
```
Comment (camilbancioiu) [inline arwen/contexts/metering_test.go:392]: Do not call `output.GetVMOutput()` after calling `metering.UpdateGasStateOnSuccess()`.

**Review by camilbancioiu — CHANGES_REQUESTED** [review]

File: arwen/contexts/metering_test.go (hunk 4 above)
Comment (alyn509) [reply arwen/contexts/metering_test.go:383]: this is not possible because testcommon imports contexts

**Review by matei-p — APPROVED** [review]

File: arwen/contexts/metering_test.go [hunk 6]
```diff
@@ -263,96 +272,259 @@ func TestMeteringContext_GasUsed_NoStacking(t *testing.T) {
...
+	require.Equal(t, childInput.GasProvided, gasUsed)
+
+	gasRemaining = math.SubUint64(parentInput.GasProvided, gasSpentByContract)
+	// calculate gas remaining
+
+	require.Equal(t, int(gasRemaining), int(metering.GasLeft()))
+}
+
+func zeroCodeCosts(context *meteringContext) {
+	//context.GasSchedule().BaseOperationCost.CompilePerByte = 0
```
Comment (sasurobert) [inline arwen/contexts/metering_test.go:479]: delete commented lines

**Review by sasurobert — APPROVED** [review]

**Review by matei-p — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by matei-p — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Pull request merged**